src/holofuel/pages/TransactionHistory/TransactionHistory.js

```diff
@@ -29,45 +30,33 @@ function useCancel () {
   })
 }
 
-function useFetchCounterparties () {
+function useTransactionsWithCounterparties () {
+  const { data: { holofuelUser: whoami = {} } = {} } = useQuery(HolofuelUserQuery)
+  const { data: { holofuelHistoryCounterparties = [] } = {} } = useQuery(HolofuelHistoryCounterpartiesQuery)
   const { data: { holofuelCompletedTransactions = [] } = {} } = useQuery(HolofuelCompletedTransactionsQuery)
   const { data: { holofuelWaitingTransactions = [] } = {} } = useQuery(HolofuelWaitingTransactionsQuery)
-  const { data: { holofuelHistoryCounterparties } = {}, client } = useQuery(HolofuelHistoryCounterpartiesQuery)
-
-  if (holofuelHistoryCounterparties) {
-    const filterTransactionsByAgentId = (agent, txListType) => txListType.filter(transaction => transaction.counterparty.id === agent.id)
-    const updateTxListCounterparties = (txListType, counterpartyList) => counterpartyList.map(agent => {
-      const matchingTx = filterTransactionsByAgentId(agent, txListType)
-      return matchingTx.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })
-    })
-
-    // Cache Write/Update for HolofuelCompletedTransactionsQuery
-    const newCompletedTxList = flatten(updateTxListCounterparties(holofuelCompletedTransactions, holofuelHistoryCounterparties))
-    client.writeQuery({
-      query: HolofuelCompletedTransactionsQuery,
-      data: {
-        holofuelCompletedTransactions: newCompletedTxList
-      }
-    })
-
-    // Cache Write/Update for HolofuelWaitingTransactionsQuery
-    const newWaitingTxList = flatten(updateTxListCounterparties(holofuelWaitingTransactions, holofuelHistoryCounterparties))
-    client.writeQuery({
-      query: HolofuelWaitingTransactionsQuery,
-      data: {
-        holofuelWaitingTransactions: newWaitingTxList
-      }
-    })
+
+  const updateCounterparties = (transactions, counterparties) => transactions.map(transaction => ({
+    ...transaction,
+    counterparty: counterparties.find(counterparty => counterparty.id === transaction.counterparty.id) || transaction.counterparty
+  }))
```

Comment:
Nice way of reworking the code to leverage the calls we are already making.  I'm glad to see we didn't fully return to an external renderNickname component.

Also, I know the fetch for counterparty nicknames has now been re-worked closer to its original form, but during optimization stages, let's consider Apollo's Read/Write Fragments (https://www.apollographql.com/docs/react/data/local-state/#writefragment-and-readfragment), which can be triggered them from within the holofuelCounterparty query resolver.  This could be a good way to still populate the Transaction schema with the counterparty nickname while reducing call frequency to `getCounterparty`.

Reply:
sounds good